test/unit/services/ledger/domain/legacy-entry-builder.spec.ts

```diff
@@ -275,20 +284,22 @@ describe("LegacyEntryBuilder", () => {
           })
           const result = reconstructEntryFromTransactions(initialResult)
 
-          expectEntryToEqual(result.debits[lndLedgerAccountId], btcAmount)
-          expect(result.credits[creditorAccountId]).toBeUndefined()
+          const credits = result.transactions.filter((t) => t.credit > 0)
+          const debits = result.transactions.filter((t) => t.debit > 0)
```

Comment:
This masks if there are zero-amount transactions in the entry. We will need a method like the following:
```
  const expectNoZeroAmounts = (entry: MediciEntry) => {
    const zeroAmounts = entry.transactions.filter((t) => t.debit === 0 && t.credit === 0)
    expect(zeroAmounts.length).toBe(0)
  }
```

and a check like:
```
expectNoZeroAmounts(result)
```

Reply:
Note; correcting this test breaks the LegacyBuilder unit tests, and we need a follow-up fix where we apply `return removeZeroAmountEntries(entry)` at lines `164` and `187` in `src/services/ledger/domain/legacy-entry-builder.ts`